### PR TITLE
EN-2131 Modified content model references detection logic, added tests 

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
Retest this please

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelManager.java [hunk 1]
```diff
@@ -190,6 +191,134 @@ public ContentModel getContentModel(long contentModelId) {
         return models;
     }
 
+    /**
```
Comment (eugeniosant) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelManager.java]: A lot of business logic added in this class, add unit tests

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelReference.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (eugeniosant) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelReference.java]: Check the right license

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelService.java [hunk 3]
```diff
@@ -224,6 +226,16 @@ public void removeContentModel(Long modelId) {
         return dtoReferences;
     }
 
+    @Override
```
Comment (eugeniosant) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentmodel/ContentModelService.java]: needs unit tests?

**Comment by entando-service** [comment]
All is Well

**Pull request merged**